lib/ngx/re.lua

```diff
@@ -236,7 +238,9 @@ function _M.split(subj, regex, opts, ctx, max, res)
 
                 sub_idx = to
 
-                if sub_idx >= len then
+                if sub_idx > len then
+                    break
+                elseif sub_idx == len and last_empty_match then
```

Comment:
With the latest patch and the addition of the trailing removing loop, those branches should not be necessary anymore. Applying this patch to your PR runs successfully as well:
```diff
diff --git a/lib/ngx/re.lua b/lib/ngx/re.lua
index 06b5e9b..6e0602f 100644
--- a/lib/ngx/re.lua
+++ b/lib/ngx/re.lua
@@ -201,8 +201,6 @@ function _M.split(subj, regex, opts, ctx, max, res)

                 if sub_idx > len then
                     break
-                elseif sub_idx == len and last_empty_match then
-                    break
                 end
             end
         end
@@ -240,8 +238,6 @@ function _M.split(subj, regex, opts, ctx, max, res)

                 if sub_idx > len then
                     break
-                elseif sub_idx == len and last_empty_match then
-                    break
                 end
             end
         end
```